### ISSUE TITLE
Does gpu.cpp support f64 or i64?

### ISSUE THREAD

**Issue by ghostplant** [AnswerDotAI/gpu.cpp#45]
I wrote some kernels based on this framework, and I found datatypes for f64 and i64 are not defined.

**Comment by MichealReed** [comment]
Looks like float32 and halfs, maybe a contribution could be added to https://github.com/AnswerDotAI/gpu.cpp/tree/main/numeric_types for f64?

**Comment by ghostplant** [comment]
I don't think so. Unlike f16, datatype f64 is not defined in dawn so maybe more contribution should be in dawn?

**Comment by MichealReed** [comment]
Thanks for investigating. Maybe you could reach out to the author here to ask about contributing this? 

https://developer.chrome.com/blog/new-in-webgpu-120

Dawn uses Gerrit here

https://dawn-review.googlesource.com/q/status:open+-is:wip

I have not even begun to digest their roadmap though.

**Comment by MichealReed** [comment]
Well, seems wgsl doesn't support f64 yet? https://webgpufundamentals.org/webgpu/lessons/webgpu-wgsl-function-reference.html
